src/l500/l500-depth.cpp

```diff
@@ -191,6 +191,31 @@ namespace librealsense
         //register_option(static_cast<rs2_option>(RS2_OPTION_DEPTH_INVALIDATION_ENABLE), _depth_invalidation_option);
     }
 
+    ivcam2::intrinsic_params l500_depth_sensor::get_intrinsic_params(const uint32_t width, const uint32_t height, ivcam2::intrinsic_depth intrinsic)
+    {
+        auto num_of_res = intrinsic.resolution.num_of_resolutions;
+
+        if (num_of_res <= MAX_NUM_OF_DEPTH_RESOLUTIONS)
```

Comment:
This is different now:
Before, it would use as many resolutions as it got. But now it won't even get in here if we get a number of resolution of 6, for example?
Do we throw if we get more resolutions back than we allow?

Reply:
Before it would use 2 real resolutions and garbage after.
The current implementation is more reliable.
If the FW will send more than 5 resolutions we will know it immediately, this kind of change require integration like we do today (the first run with alert us with the problem).
We want to protect LRS from exceptions and undefined behavior.
we still allow the count to grow from 3 to 5.
we can set the max size to higher from 5 but it will be a waist of memory for something that might happen one day and might not.

What do you think?